### DISCUSSION TITLE
how can i use useConfigure to pass in custom filters

### DISCUSSION THREAD

**Discussion by weiklr** [algolia/instantsearch#5645] [Q&A]
I'm trying to pass in filters dynamically to instant search and i thought i could useConfigure hook to do this. the useConfigure and pass in a set of filters dynamically. (e.g. someone clicks "Product" button on my ui and my filter becomes filter:"product:widget")

I noticed that UseConfigureProps only expose PlainSearchParameters. But we also have a SearchParameters type. Should UseConfigureProps use SearchParameters type instead?

I was trying to follow this example to pass in filters, but it doesn't work as Typescript says it can't find filters in UseConfigureProps. Is this a bug?

https://www.algolia.com/doc/api-reference/widgets/configure/react-hooks/

**Comment by javierdebug** [comment]
You can create a variable and dynamically create the filters' string according to the user's selections and then pass it to the `filters` props in the `Configure` (see example: https://www.algolia.com/doc/api-reference/widgets/configure/react/#examples). This is a `string` type, so TS should work.
